app/controllers/users_controller.rb

```diff
@@ -18,8 +21,33 @@ def new
     @user = User.new
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
```

Comment:
you should define 
`@user = User.find_by id: params[:id]`
in a method.
and call it by before_action
```
before_action :load_user, only: [:edit, :update, :destroy]
def load_user
  @user = User.find_by id: params[:id]
  redirect_to root_path unless @user
end
```
so in  update, edit, destroy, dont need to find user any more